aignostic/aignostic/metrics/metrics.py

```diff
@@ -10,17 +10,22 @@
 
 def calculate_metrics(y_true, y_pred, metrics):
     """
-        Calculate the metrics for the given y_true and y_pred
-        Args:
-            y_true: numpy array of true labels
-            y_pred: numpy array of predicted labels
-            metrics: list of metric functions e.g. "accuracy", "precision"
-    """
-    results = {}
-    for metric in metrics:
-        results[metric] = globals()[metric](y_true, y_pred)
+    Calculate the metrics for the given y_true and y_pred
 
-    return results
+    Params:
+        y_true: list of true labels
+        y_pred: list of predicted labels
+        metrics: list of metric functions e.g. "accuracy", "precision"
+    """
+    try:
+        results = {}
+        for metric in metrics:
+            results[metric] = globals()[metric](np.array(y_true), np.array(y_pred))
+
+        return results
+    except Exception as e:
```

Comment:
Note that this will also catch Keyboard interrupts as well, which you probably don't want (makes program more difficult to terminate); though I can't remember the better way to do it (normally you pick a more specific error type)